presto-main/src/main/java/com/facebook/presto/operator/aggregation/QuantileDigestAggregationFunction.java

```diff
@@ -13,129 +13,87 @@
-import com.facebook.presto.spi.block.BlockBuilder;
-import com.facebook.presto.spi.type.QuantileDigestType;
+import com.facebook.presto.StatisticalDigest;
+import com.facebook.presto.operator.aggregation.state.StatisticalDigestState;
+import com.facebook.presto.operator.aggregation.state.StatisticalDigestStateFactory;
+import com.facebook.presto.spi.PrestoException;
 import com.facebook.presto.spi.type.StandardTypes;
 import com.facebook.presto.spi.type.Type;
-import com.facebook.presto.spi.type.TypeManager;
 import com.facebook.presto.spi.type.TypeSignature;
-import com.facebook.presto.spi.type.TypeSignatureParameter;
-import com.google.common.collect.ImmutableList;
+import com.facebook.presto.type.StatisticalQuantileDigest;
 import io.airlift.stats.QuantileDigest;
 
 import java.lang.invoke.MethodHandle;
-import java.util.List;
-import java.util.stream.Collectors;
 
-import static com.facebook.presto.operator.aggregation.AggregationMetadata.AccumulatorStateDescriptor;
-import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata;
-import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.INPUT_CHANNEL;
-import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.STATE;
-import static com.facebook.presto.operator.aggregation.AggregationUtils.generateAggregationName;
 import static com.facebook.presto.operator.aggregation.FloatingPointBitsConverterUtil.doubleToSortableLong;
 import static com.facebook.presto.operator.aggregation.FloatingPointBitsConverterUtil.floatToSortableInt;
 import static com.facebook.presto.operator.scalar.QuantileDigestFunctions.DEFAULT_ACCURACY;
 import static com.facebook.presto.operator.scalar.QuantileDigestFunctions.DEFAULT_WEIGHT;
 import static com.facebook.presto.operator.scalar.QuantileDigestFunctions.verifyAccuracy;
 import static com.facebook.presto.operator.scalar.QuantileDigestFunctions.verifyWeight;
-import static com.facebook.presto.spi.function.Signature.comparableTypeParameter;
-import static com.facebook.presto.spi.type.BigintType.BIGINT;
-import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static com.facebook.presto.spi.StandardErrorCode.NOT_SUPPORTED;
+import static com.facebook.presto.spi.type.StandardTypes.QDIGEST;
 import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
 import static com.facebook.presto.util.Reflection.methodHandle;
-import static com.google.common.collect.ImmutableList.toImmutableList;
 import static java.lang.Float.intBitsToFloat;
 import static java.lang.String.format;
 import static java.lang.invoke.MethodHandles.insertArguments;
 
 public final class QuantileDigestAggregationFunction
-        extends SqlAggregationFunction
+        extends AbstractStatisticalDigestAggregationFunction
```

Comment:
We don't typically mark our abstract classes as abstract.  I would name it simply (`StatisticalDigestAggregationFunction`, or following advice here:

https://github.com/prestodb/presto/pull/13006/files#r301640507

Even `QuantileSketchAggregationFunction`